mh/quest_utils.py

```diff
@@ -1441,10 +1471,14 @@ def generate_flags(byte1, byte2, byte3, byte4):
 
     return res
 
+
 def read_quest_sm_data(fname):
-    with open('event/'+fname) as f:
-        data = f.read()
-    return bytearray.fromhex(data.replace(" ","").replace("\n",""))
+    if path.exists('event/'+fname):
+        with open('event/'+fname) as f:
+            data = f.read()
+        return bytearray.fromhex(data.replace(" ","").replace("\n",""))
+    else:
+        return None
```

Comment:
By inverting the condition you can lower the indentation level and save lines. `os.path.join` should be used to concatenate path instead of the `+` operator.

_(the following suggestion was untested, the `open`'s mode might need to be adjusted)_
```suggestion
    data_path = path.join("event", fname)
    if not path.exists(data_path):
        return None
    with open(data_path, "rb") as f:
        data = f.read()
    return bytearray.fromhex(data.replace(b" ", b"").replace(b"\n", b""))
```

Reply:
`open` should not have the `b` mode, since the file it's reading is hex in plaintext. (I'm aware that plaintext hex is not a good way of storing it, but it makes testing easier. Once small monster data is properly reverse engineered this will be switched to something more appropriate.)

Reply:
Fixed